lib/buffer.js

```diff
@@ -246,18 +247,30 @@ function fromArrayLike(obj) {
   const length = obj.length;
   const b = allocate(length);
   for (var i = 0; i < length; i++)
-    b[i] = obj[i] & 255;
+    b[i] = obj[i];
   return b;
 }
 
 function fromArrayBuffer(obj, byteOffset, length) {
+  if (!isArrayBuffer(obj))
+    throw new TypeError('argument is not an ArrayBuffer');
```

Comment:
> the expected TypeError is thrown when accessing byteLength below, although with a different message (which would technically make dropping this a semver-major change…)

Yes, that's what I meant by not passing the test. Anyway, I totally don't mind changing it if that's what others want.